packages/api/src/resolvers/integrations/index.ts

```diff
@@ -0,0 +1,118 @@
+import { authorized } from '../../utils/helpers'
+import {
+  MutationSetIntegrationArgs,
+  SetIntegrationError,
+  SetIntegrationErrorCode,
+  SetIntegrationSuccess,
+} from '../../generated/graphql'
+import { getRepository } from '../../entity/utils'
+import { User } from '../../entity/user'
+import { Integration } from '../../entity/integration'
+import { analytics } from '../../utils/analytics'
+import { env } from '../../env'
+import { validateToken } from '../../services/integrations'
+import { deleteTask, enqueueSyncWithIntegration } from '../../utils/createTask'
+import { AppDataSource } from '../../server'
+
+export const setIntegrationResolver = authorized<
+  SetIntegrationSuccess,
+  SetIntegrationError,
+  MutationSetIntegrationArgs
+>(async (_, { input }, { claims: { uid }, log }) => {
+  log.info('setIntegrationResolver')
+
+  try {
+    const user = await getRepository(User).findOneBy({ id: uid })
+    if (!user) {
+      return {
+        errorCodes: [SetIntegrationErrorCode.Unauthorized],
+      }
+    }
+
+    const integrationToSave: Partial<Integration> = {
```

Comment:
Do we need to sanitize these input values? I guess GQL kind of does already, and then the DB will verify that its a valid type.